src/xdp/apikernel.c

```diff
@@ -316,6 +304,30 @@ XskIoctlKernel(
     return STATUS_NOT_SUPPORTED;
 }
 
+static
+XDP_STATUS
+XdpCloseHandleKernel(
+    _In_ HANDLE Handle
+    )
+{
+    XDP_FILE_OBJECT_HEADER *FileObjectHeader = (XDP_FILE_OBJECT_HEADER *)Handle;
+    switch (FileObjectHeader->ObjectType) {
```

Comment:
The idea here is to create a function pointer and place it within the file object header's dispatch table (XDP_FILE_DISPATCH)

Reply:
Looks like the dispatch table is ready, but problem is how I can call from this API.
Functions in XDP_FILE_DISAPATCH requires `(IRP *irp, IO_STACK_LOCATION *IrpSp)` as arguments.
Where are these arguments taken? or any other callback automatically fills them? if so, what should be called in this APi?

Reply:
Yup, to clarify, my suggestion is to add one more function pointer type to the dispatch table, and it takes just the FILE_OBJECT_HEADER pointer as a parameter, rather than IRPs.